bench/lib/display.ml

```diff
@@ -0,0 +1,60 @@
+(*
+  This file is part of scilla.
+  Copyright (c) 2018 - present Zilliqa Research Pvt. Ltd.
+  scilla is free software: you can redistribute it and/or modify it under the
+  terms of the GNU General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option) any later
+  version.
+  scilla is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+  A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License along with
+  scilla.  If not, see <http://www.gnu.org/licenses/>.
+*)
+
+open Core
+open Textutils
+
+let print_tests groups =
+  let open Core_bench in
+  let open Ascii_table in
+  List.iter groups ~f:(fun group ->
+      let cells =
+        List.map (Test.tests group)
+          ~f:(fun test ->
+              [ Test.name group
+              ; Test.Basic_test.name test
+              ])
+      in simple_list_table
+        ~display:Display.column_titles
+        ["group"; "test"] cells)
+
+let to_ms = Util.ns_to_ms_string
+
+let print_results results =
+  let open Ascii_table in
+  let open Measurement_result in
+  let cells = List.map results ~f:(fun res ->
+      let ms = to_ms res.time_per_run_nanos in
+      [res.benchmark_name; ms])
+  in simple_list_table
+    ~display:Display.column_titles
+    ["benchmark name"; "ms"] cells
+
+let print_comparison
+    ~previous:(prev, prev_ts) ~current:(curr, curr_ts) ~deltas =
+  let open Ascii_table in
+  let open Measurement_result in
+  let cells = List.map3_exn prev curr deltas ~f:(fun prev curr delta ->
+      [ curr.benchmark_name
+      ; to_ms prev.time_per_run_nanos
+      ; to_ms curr.time_per_run_nanos
+      ; to_ms delta.time_per_run_nanos
+      ])
+  in simple_list_table
+    ~display:Display.column_titles
+    [ "benchmark name"
+    ; "previous (ms)"
+    ; "current (ms)"
+    ; sprintf "delta (%s / %s)" prev_ts curr_ts
```

Comment:
Is there anything else that can be done here as opposed to have an output like this?
```
                        benchmark name   previous (ms)   current (ms)   delta (20191203074607 / 20191203074848)  
1810 ╶────────────────────────────────────╴ ╶─────────────╴ ╶────────────╴ ╶───────────────────────────────────────╴ 
1811                         exp/eval/addr         37.94ms        37.77ms                                 -169.16us  
1812
```
The timestamps IMHO are not very readable. Perhaps commit trimmed commit hashes or just `before` / `after`.
Also, it would be nice to see not only absolute diff, but a relative too (by default).

Reply:
Maybe just `delta` would suffice?

The idea behind displaying timestamps was that it is easier to understand what is compared. In addition to the `-ci` mode, there is still the opportunity to run these benchmarks locally a few times during development without making commits. Thats why results of each run are not tied to the hash of the commit and are saved in a directory named after timestamp. This makes it possible, for example, to compare the results of the current run with the results of one of the previous runs by specifying its timestamp with the `-timestamp` flag.

Reply:
Could you please explain the difference between absolute vs relative diff?
(In the current implementation `delta = after - before`). Thanks!

Reply:
Sure, let's not tie that to commit hashes --  just `delta (ms)` or something like that would be fine. A relative change would be something like `(after - before) / before * 100 %`.